los/los.c

```diff
@@ -1,141 +1,107 @@
 #include "los.h"
-#include <stdio.h>
-#include <math.h>
-
-#define TRUE 1
-#define FALSE 0
-
-void step_towards(int *x, int *y, int target_x, int target_y, int diagonal) {
-    int dx = target_x - *x;
-    int dy = target_y - *y;
-
-    if (abs(dx) > abs(dy)) {
-        // need to decrease x
-        *x += (*x < target_x ? 1 : -1);
-    } else if (abs(dx) < abs(dy)) {
-        // need to decrease y
-        *y += (*y < target_y ? 1 : -1);
-    } else if (rand() % 2 == 0) {
-        // pick at random
-        // TODO Sometimes mobs will bump into a wall when they don't need to
-        *x += (*x < target_x ? 1 : -1);
-    } else {
-        *y += (*y < target_y ? 1 : -1);
-    }
+
+bool line_of_sight(level *lvl, int a_x, int a_y, int b_x, int b_y) {
+    //TODO how the hell is is_position_valid working in here?
+    //TODO because I included level.h? yuck!
```

Comment:
This is just like any function call. Why is that yuck?

Reply:
Circular includes? I still don't understand what sort of complicated include webs I'm allowed to build.

Reply:
Sounds good to me.